apiserver/test/e2e/utils.go

```diff
@@ -93,13 +102,62 @@ func waitForRunningCluster(t *testing.T, tCtx *End2EndTestingContext, clusterNam
 
 func waitForDeletedCluster(t *testing.T, tCtx *End2EndTestingContext, clusterName string) {
 	// wait for the cluster to be deleted
-	err := wait.PollUntilContextTimeout(tCtx.ctx, 500*time.Millisecond, 3*time.Minute, false, func(_ context.Context) (done bool, err error) {
-		_, err = tCtx.GetRayClusterByName(clusterName)
+	g := gomega.NewWithT(t)
+	g.Eventually(func() bool {
+		_, err := tCtx.GetRayClusterByName(clusterName)
 		if err != nil && strings.Contains(err.Error(), "rayclusters.ray.io \""+tCtx.GetRayClusterName()+"\" not found") {
-			return true, nil
+			return true
 		}
 		t.Logf("Found ray cluster '%s'", clusterName)
-		return false, err
-	})
-	require.NoErrorf(t, err, "No error expected when deleting ray cluster: '%s', err %v", clusterName, err)
+		return false
+	}, TestTimeoutMedium, TestPollingInterval).Should(gomega.BeTrue())
+}
+
+func waitForRayJob(t *testing.T, tCtx *End2EndTestingContext, rayJobName string, expectedJobStatuses []rayv1api.JobStatus) {
+	// `expectedJobStatuses` is a slice of job statuses that we expect the job to be in
+	// wait for the job to be in any of the `expectedJobStatuses` state for 3 minutes
+	// if is not in that state, return an error
+	g := gomega.NewWithT(t)
+
+	g.Eventually(func() bool {
+		rayJob, err := tCtx.GetRayJobByName(rayJobName)
+		if err != nil {
+			t.Logf("Error getting ray job '%s': %v", rayJobName, err)
+			return false
+		}
+		t.Logf("Waiting for ray job '%s' to be in one of the expected conditions %s", rayJobName, expectedJobStatuses)
+		return slices.Contains(expectedJobStatuses, rayJob.Status.JobStatus)
+	}, TestTimeoutMedium, TestPollingInterval).Should(gomega.BeTrue())
+}
+
+func waitForDeletedRayJob(t *testing.T, tCtx *End2EndTestingContext, jobName string) {
+	// wait for the job to be deleted
+	// if is not in that state, return an error
+	g := gomega.NewWithT(t)
+	t.Logf("Starting to wait for ray job %s to be deleted", jobName)
+	g.Eventually(func() error {
+		rayJob, err := tCtx.GetRayJobByName(jobName)
+		if err != nil {
+			return err
+		}
+		t.Logf("Find rayJob with status %v", rayJob.Status.JobStatus)
+		return nil
+	}, TestTimeoutMedium, TestPollingInterval).Should(gomega.MatchError("rayjobs.ray.io \"" + jobName + "\" not found"))
+
+	t.Logf("Ray job %s successfully deleted", jobName)
+}
+
+func waitForDeletedService(t *testing.T, tCtx *End2EndTestingContext, serviceName string) {
```

Comment:
```suggestion
func waitForServiceToDisappear(t *testing.T, tCtx *End2EndTestingContext, serviceName string) {
```
Nit: These `waitForDeletedXXX` helpers don't really have to be tight with deletions. We can make them more flexible by changing some wording.

Reply:
Thanks RueiAn !  Renamed!

Reply:
The logs in those functions should also be changed accordingly.